src/commands/synthetics/test.ts

```diff
@@ -15,10 +16,21 @@ export const getTestConfigs = async (
 
   suites.push(...suitesFromFiles)
 
+  // TODO SYNTH-12989: Clean up deprecated `config` in favor of `testOverrides`
+  const isUsingConfig = suites.some((suite) =>
+    suite.content.tests.some((test) => Object.keys(test.config ?? {}).length > 0)
+  )
+  if (isUsingConfig) {
+    reporter?.error(
+      "The 'config' property is deprecated. Please use 'testOverrides' instead.\nIf both 'config' and 'testOverrides' properties exist, 'testOverrides' is used!\n"
+    )
+  }
```

Comment:
I understand that you can't log the warning in `replaceConfigWithTestOverrides()` because it's in a loop. But I'd like that to have everything about compatibility (logic, and generated warnings) in a single file.

You could move this into a `warnIfDeprecatedConfigUsed()`:

```suggestion
  warnIfDeprecatedConfigUsed(...)
```

Reply:
Yeah i originally had a function in compatibility.ts doing that, but then moved some stuff around and i didn't like how it looked. I guess in this last iteration it is possible to extract it again though 👍